src/Compilers/CSharp/Test/Emit3/Semantics/CollectionExpressionTests_WithElement_Constructor.cs

```diff
@@ -333,7 +361,218 @@ static void Main()
+
+        var semanticModel = compilation.GetSemanticModel(compilation.SyntaxTrees.Single());
+        var root = semanticModel.SyntaxTree.GetRoot();
+
+        var withElements = root.DescendantNodes().OfType<WithElementSyntax>().ToArray();
+        Assert.Equal(4, withElements.Length);
+
+        var constructor1 = (IMethodSymbol?)semanticModel.GetSymbolInfo(withElements[0]).Symbol;
+        var constructor2 = (IMethodSymbol?)semanticModel.GetSymbolInfo(withElements[1]).Symbol;
+        var constructor3 = (IMethodSymbol?)semanticModel.GetSymbolInfo(withElements[2]).Symbol;
+        var constructor4 = (IMethodSymbol?)semanticModel.GetSymbolInfo(withElements[3]).Symbol;
+
+        Assert.NotNull(constructor1);
+        Assert.NotNull(constructor2);
+        Assert.NotNull(constructor3);
+        Assert.NotNull(constructor4);
+
+        Assert.Equal(constructor1, constructor2);
+        Assert.Equal(constructor1, constructor3);
+        Assert.Equal(constructor1, constructor4);
+
+        Assert.Equal("MyList", constructor1.ContainingType.Name);
+
+        Assert.True(constructor1.Parameters is [{ Name: "capacity", Type.SpecialType: SpecialType.System_Int32 }, { Name: "name", Type.SpecialType: SpecialType.System_String }]);
+
+        var operation = semanticModel.GetOperation(root.DescendantNodes().OfType<LocalDeclarationStatementSyntax>().ToArray()[1]);
+        VerifyOperationTree(compilation, operation, """
+            IVariableDeclarationGroupOperation (1 declarations) (OperationKind.VariableDeclarationGroup, Type: null) (Syntax: 'MyList<int> ... y: 10), 2];')
+               IVariableDeclarationOperation (1 declarators) (OperationKind.VariableDeclaration, Type: null) (Syntax: 'MyList<int> ... ty: 10), 2]')
+                 Declarators:
+                     IVariableDeclaratorOperation (Symbol: MyList<System.Int32> list2) (OperationKind.VariableDeclarator, Type: null) (Syntax: 'list2 = [wi ... ty: 10), 2]')
+                       Initializer:
+                         IVariableInitializerOperation (OperationKind.VariableInitializer, Type: null) (Syntax: '= [with(cap ... ty: 10), 2]')
+                           IConversionOperation (TryCast: False, Unchecked) (OperationKind.Conversion, Type: MyList<System.Int32>, IsImplicit) (Syntax: '[with(capacity: 10), 2]')
+                             Conversion: CommonConversion (Exists: True, IsIdentity: False, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+                             Operand:
+                               ICollectionExpressionOperation (1 elements, ConstructMethod: MyList<System.Int32>..ctor([System.Int32 capacity = 0], [System.String name = "default"])) (OperationKind.CollectionExpression, Type: MyList<System.Int32>) (Syntax: '[with(capacity: 10), 2]')
+                                 Elements(1):
+                                     ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 2) (Syntax: '2')
```

Comment:
Should the "capacity" argument be captured anywhere in this `ICollectionExpressionOperation`?

Reply:
i think we'd have to consider expanding on ICollectionExpressionOp.  I'm not proposing such a thing.

Reply:
From discussion with @AlekseyTs, confirmed that we'll want the argument operations to show up in the operation tree and any branching from them to show up in CFG. Added a note to test plan (this doesn't need to be done in this PR)